modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb

```diff
@@ -41,28 +44,46 @@ def submit_supporting_documents
 
       private
 
-      def get_file_paths_and_metadata(parsed_form_data)
+      def get_attachment_ids_and_form(parsed_form_data)
         form_id = get_form_id
         form = "IvcChampva::#{form_id.titleize.gsub(' ', '')}".constantize.new(parsed_form_data)
-        filler = IvcChampva::PdfFiller.new(form_number: form_id, form:)
+        attachment_ids = [form_id]
+
+        if form_id == 'vha_10_10d'
+          parsed_form_data['applicants'].each do |applicant|
+            next unless applicant.key?('applicant_supporting_documents')
+
+            applicant['applicant_supporting_documents'].each do |documents|
+              documents.each do |document|
+                attachment_ids << document['attachment_id']
+              end
+            end
+          end
+        end
+        [attachment_ids, form]
+      end
 
+      def get_file_paths_and_metadata(parsed_form_data)
+        attachment_ids, form = get_attachment_ids_and_form(parsed_form_data)
+        filler = IvcChampva::PdfFiller.new(form_number: form.form_id, form:)
         file_path = if @current_user
                       filler.generate(@current_user.loa[:current])
                     else
                       filler.generate
                     end
-
         metadata = IvcChampva::MetadataValidator.validate(form.metadata)
         file_paths = form.handle_attachments(file_path)
 
-        [file_paths, metadata]
+        [file_paths, metadata, attachment_ids]
       end
 
       def get_form_id
         form_number = params[:form_number]
         raise 'missing form_number in params' unless form_number
 
-        FORM_NUMBER_MAP[form_number]
+        form_number_without_colon = form_number.sub(':', '')
```

Comment:
Can you explain why we have to add this? The form map key/value above doesn't have colons.

Reply:
For 1010D it does.

Reply:
Really? The map is `'10-10D' => 'vha_10_10d',` when does the colon come into play?